src/typescript/frontend/src/utils/geolocation.ts

```diff
@@ -4,17 +4,29 @@ import { GEOBLOCKED, GEOBLOCKING_ENABLED } from "lib/server-env";
 import { headers } from "next/headers";
 
 export type Location = {
-  countryCode: string;
-  regionCode: string;
+  countryCode: string | null;
+  regionCode: string | null;
 };
 
 const isDisallowedLocation = ({ countryCode, regionCode }: Location) => {
-  if (GEOBLOCKED.countries.includes(countryCode)) {
+  if (countryCode && GEOBLOCKED.countries.includes(countryCode)) {
     return true;
   }
-  const isoCode = `${countryCode}-${regionCode}`;
-  if (GEOBLOCKED.regions.includes(isoCode)) {
-    return true;
+  if (regionCode) {
+    const isoCode = `${countryCode}-${regionCode}`;
+    if (GEOBLOCKED.regions.includes(isoCode)) {
+      return true;
+    }
+  }
+  if (countryCode && !regionCode) {
+    if (GEOBLOCKED.regions.map((r) => r.split("-")[0]).includes(countryCode)) {
+      return true;
+    }
+  }
+  if (!countryCode && regionCode) {
+    if (GEOBLOCKED.countries.includes(regionCode.split("-")[0])) {
```

Comment:
```suggestion
  if (!countryCode && regionCode) {
    // Note that even if the `regionCode` is `XX`, and `XX` is a banned country, this will return
    // `true` and thus block the user, because "XX".split("-")[0] is just "XX".
    if (GEOBLOCKED.countries.includes(regionCode.split("-")[0])) {
```